packages/utils/src/date/Locale.ts

```diff
@@ -209,90 +231,143 @@ class Locale {
     const eraNames: FormattedNames[] = [];
     const dates = [utcParse('-0001-01-01').getTime(), Date.UTC(2018, 0, 1)];
     for (let i = 0; i <= 1; i += 1) {
-      const era = dates[i];
-      const part = this.formatter.formatToParts(era).find(part => part.type === 'era');
-      if (part) {
+      const value = dates[i];
+      const era = this.formatter.formatToParts(value).find(part => part.type === 'era');
+      if (era) {
         eraNames.push({
-          value: era,
-          name: part.value.toLowerCase()
+          value,
+          name: era.value.toLowerCase()
         });
       }
     }
     return eraNames;
   }
 
-  /**
-   * Escape list with literals to be used with RegExp
-   * @param list A list with items
-   * @returns escaped list
-   */
-  private escapeList (list: FormattedNames[]): string {
-    // There might be duplicates, which are not required in RegExp
-    return `(${[...new Set(list.map(item => item.name))].map(item => regExpEscape(item)).join('|')})`;
-  }
-
   private _regExp: RegExp | null = null;
   /**
    * Get regular expression for the format
    */
   private get regExp (): RegExp {
     if (this._regExp) {
+      this._regExp.lastIndex = 0; // 0 index?
```

Comment:
Better comment please. I have no idea what this means :)

Reply:
Comment removed